src/koinos/chain/controller.cpp

```diff
@@ -322,6 +322,8 @@ apply_block_result controller_impl::apply_block( const protocol::block& block, c
     KOINOS_ASSERT( block_id == root->id(), unknown_previous_block_exception, "unknown previous block" );
     return {}; // Block is current LIB
   }
+  else
+    KOINOS_ASSERT( parent_node->is_finalized(), unknown_previous_block_exception, "unknown previous block" );
```

Comment:
If the `if` uses braces we should use braces in the `else`

Reply:
Also, it appears there's an integration failure

Reply:
It fails the integration test because the branch was based on a commit prior to the nonce changes. I built the image manually and pushed it for testing p2p without all of the nonce code.

I can merge master in to the branch to pass integration tests, but then I cannot test it on its own with p2p.